lib5k/src/main/java/io/github/frc5024/purepursuit/pathgen/BezierPath.java

```diff
@@ -0,0 +1,214 @@
+package io.github.frc5024.purepursuit.pathgen;
+
+import java.util.ArrayList;
+import ca.retrylife.ewmath.MathUtils;
+import edu.wpi.first.wpilibj.geometry.Translation2d;
+import edu.wpi.first.wpilibj.util.Units;
+
+/**
+ * This class is used to generate points along a bezier curves
+ */
+public class BezierPath extends Path{
+
+	// Maximum space allowed between points
+	private double maxSeperation = Units.inchesToMeters(4);
+
+	// The lookup table
+	private static ArrayList<int[]> binomialLookUpTable;
+
+	/**
+	 * 
+	 * @param wayPoints The 3 way points of a quadratic bezier curve
+	 */
+	public BezierPath(Translation2d[] wayPoints) {
+		this(wayPoints, new double[] { 1, 1, 1 }, Units.inchesToMeters(4));
+	}
+
+	/**
+	 * 
+	 * @param wayPoints The waypoints of the bezier curve
+	 * @param weights   The weights for each point point 0 and 2 should stay as
+	 *                  close to 1 as possible
+	 */
+	public BezierPath(Translation2d[] wayPoints, double[] weights){
+		this(waypoint, weights, Units.inchesToMeters(4));
+	}
+
+	/**
+	 * 
+	 * @param wayPoints The waypoints of the bezier curve
+	 * @param weights   The weights for each point point 0 and 2 should stay as
+	 *                  close to 1 as possible
+	 * @param spacing   the maximum amount of spacing allowed
+	 */
+	public BezierPath(Translation2d[] wayPoints, double[] weights, double spacing) {
+		this.waypoints = wayPoints;
+		this.points = new ArrayList<>();
+		this.maxSeperation = spacing;
+
+		if (binomialLookUpTable == null) {
+			initilizeLookupTable();
+		}
+
+		if(wayPoints.length != weights.length){
+			weights = new double[wayPoints.length];
+
+			for(int i = 0; i < wayPoints.length; i++){
+				weights[i] = 1;
```

Comment:
I recommend copying the weights over the new array like this:

```java
double [] newWeights = new double[wayPoints.length];

for(int i = 0; i < wayPoints.length; i++){
	if(i < weights.length){
		newWeights[i] = weights[i];
	} else {
		newWeights[i] = 1;
	}
}
```

Reply:
Otherwise, the existing code might confuse someone. "Why are all my weights being ignored"? for example